Correct intellisense and indentation for fixed statements

lorcanmooney opened the pull request:
Ensures the IDE treats `fixed` statements as statement contexts so IntelliSense will be shown after the closing parenthesis.

This also adds some missing "magic" to correct the indentation of `fixed` statements. With this change, they're treated similar to `using`.

Fixes #11213.

dnfclas commented:
Hi __@lorcanmooney__, I'm your friendly neighborhood .NET Foundation Pull Request Bot (You can call me DNFBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla2.dotnetfoundation.org.
    </span>

TTYL, DNFBOT;

dnfclas commented:
__@lorcanmooney__, Thanks for signing the contribution license agreement so quickly! Actual humans will now validate the agreement and then evaluate the PR.
<br />Thanks, DNFBOT;

Pilchie commented:
Thanks for the submission, @lorcanmooney!

Quick question - did VS2013 have the indentation/formatting before for `fixed`?  I thought it was unique to `using`.  I like it, but I generally don't like it when people open existing code, format, and have it change for no reason with no option to keep the formatting the way it used to be.

Tag all of @dotnet/roslyn-ide for review.  Also, @dotnet-bot test vsi please.

lorcanmooney commented:
Yes, this change restores parity with VS2013.

For reference, given the following code...
```
					using (null)
				using (null)
			if (false)
		fixed (int* a = &i)
	fixed (int* b = &i)
if (true)
{
}
```
...this is how VS2013 formats it:
```
using (null)
using (null)
    if (false)
        fixed (int* a = &i)
        fixed (int* b = &i)
            if (true)
            {
            }
```
...and this is the current state of VS2015:
```
using (null)
using (null)
	if (false)
		fixed (int* a = &i)
	fixed (int* b = &i)
if (true)
		{
		}
```

DustinCampbell commented:
Nice! :+1:

Pilchie commented:
What about `fixed` with nested `using` and vice versa?  If those match up, then :+1:.

Pilchie commented:
retest vsi please - CSharpDebuggerIntelliSense failed, but I suspect it isn't related.  We'll give it one retry.

balajikris commented:
@dotnet-bot retest roslyn_prtest_win_vsi2 please.
// Previous failure: http://dotnet-ci.cloudapp.net/job/roslyn_prtest_win_vsi2/94/
// Retest reason: CSharpExpressionEvaluator failed. Does not look related, investigating offline.

balajikris commented:
retest roslyn_prtest_win_vsi2 please

lorcanmooney commented:
@Pilchie Yes, the `fixed`+`using` permutations behave as in VS2013. In both cases, they're indented to different levels. I'll add some additional unit tests around these variations, but they'll probably have to wait until next weekend.

@balajikris Thanks!

balajikris commented:
retest vsi2 please

balajikris commented:
@dotnet-bot retest vsi please

// Above trick of retesting only a particular queue vsi2 didn't work

balajikris commented:
:+1: for the change itself.

lorcanmooney commented:
Looks like vsi2 failed again, but I don't have access to the results. Any suggestions on how I can figure out what went wrong?

balajikris commented:
@lorcanmooney : Thanks for the note, I'll take over investigating the failure. It does seem unrelated to this change again. 
@Pilchie: what's the guidance here, do we require VSI passing before merging? Looks like we have various flakiness in VSI and I'm still in the process of flushing them all out.

balajikris commented:
@jasonmalinowski : looks like a diagnostic you added got triggered in http://dotnet-ci.cloudapp.net/job/roslyn_prtest_win_vsi2/99

Is this actionable? How to route this failure further?
~~~
12:31:10 System.Exception: The text that we thought we were saving isn't what we saved!
12:31:10    at Roslyn.VSHosting.Test.CloseFileAction.Execute(TaoEngine engine) in d:\j\workspace\roslyn_prtest_win_vsi2\Closed\Hosting\Test\Actions\VisualStudio\CloseFileAction.cs:line 60
12:31:10    at Tao.Engine.DebuggableActionExecutor.Execute() in f:\dd\src\Omni\Tao\Tao.Engine\ActionExecutor.cs:line 139
12:31:10    at Tao.Engine.DefaultActionExecutor.Execute() in f:\dd\src\Omni\Tao\Tao.Engine\ActionExecutor.cs:line 36
~~~

source: http://dotnet-ci.cloudapp.net/job/roslyn_prtest_win_vsi2/99/consoleFull#-169875419061aefe2b-268e-4e71-8c03-11b4c0acbfbc

balajikris commented:
@dotnet-bot test vsi2 please

balajikris commented:
> I'll add some additional unit tests around these variations

@lorcanmooney Are you planning to add these tests to this PR? If so, I'm okay to keep this open until you are finished.

rchande commented:
:+1:

lorcanmooney commented:
@balajikris If that's ok, then yes, I think it would be best to defer the merge so we can keep all the changes in _this_ PR. It shouldn't be much work, and I'll try to get to it in the next few days.

balajikris commented:
@lorcanmooney that's perfectly alright, thanks.

lorcanmooney commented:
I've added a handful of tests for the various special-cased combinations. I'm closing and re-opening the PR to see if I can trigger a new test pass.

lorcanmooney commented:
I think this is ready to merge now.

balajikris commented:
thanks for the PR, merging in.
